src/Nfts/ImageGallery/ImageGallery.tsx

```diff
@@ -58,81 +91,133 @@ interface ModeratedImageProps {
 
 const ModeratedImage = ({fingerprint, image, text, onPress}: ModeratedImageProps) => {
   const wallet = useSelectedWallet()
-  const moderationStatusQuery = useQuery({
-    queryKey: [wallet.id, 'nft', fingerprint],
-    queryFn: () => wallet.fetchNftModerationStatus(fingerprint),
-  })
+  const moderationStatusQuery = useNftModerationStatus({wallet, fingerprint})
 
   useEffect(() => {
     if (moderationStatusQuery.data === 'pending') {
       moderationStatusQuery.refetch()
     }
   }, [moderationStatusQuery.data, moderationStatusQuery])
+  const size = getImageSize()
+
+  const isPendingReview = moderationStatusQuery.data === 'pending' || moderationStatusQuery.data === 'manual_review'
+  const showSkeleton = moderationStatusQuery.isLoading || isPendingReview
 
-  const showSkeleton = moderationStatusQuery.isLoading
-  const isImageApproved = moderationStatusQuery.data === 'green'
+  const isGreenImage = moderationStatusQuery.data === 'green'
+  const isYellowImage = moderationStatusQuery.data === 'yellow'
+  const isRedImage = moderationStatusQuery.data === 'red'
 
   if (showSkeleton) {
     return <SkeletonImagePlaceholder />
   }
 
   return (
-    <TouchableOpacity onPress={onPress} style={styles.imageContainer}>
-      {isImageApproved ? (
+    <TouchableOpacity disabled={isRedImage} onPress={onPress} style={[styles.imageContainer]}>
+      {isGreenImage ? (
         <>
-          <Image source={{uri: image}} style={styles.image} />
-          <Spacer height={8} />
-          <Text style={styles.textTop}>{text}</Text>
-          <Spacer height={13} />
+          <Image source={{uri: image}} style={[styles.image, {width: size, height: size}]} />
+          <Spacer height={IMAGE_PADDING} />
+          <Text style={[styles.textTop, {width: size}]}>{text}</Text>
+          <Spacer height={TEXT_PADDING} />
         </>
-      ) : (
+      ) : isYellowImage ? (
+        <View>
+          <View style={styles.imageWrapper}>
+            <Image source={{uri: image}} style={[styles.image, {width: size, height: size}]} blurRadius={20} />
+            <View style={styles.eyeWrapper}>
+              <Icon.EyeOff size={20} color="#FFFFFF" />
+            </View>
+          </View>
+          <Spacer height={IMAGE_PADDING} />
+          <Text style={[styles.textTop, {width: size}]}>{text}</Text>
+          <Spacer height={TEXT_PADDING} />
+        </View>
+      ) : isRedImage ? (
         <>
-          <View style={styles.image} />
-          <Spacer height={8} />
-          <Text style={styles.textTop}>Image is not approved</Text>
-          <Spacer height={13} />
+          <Image source={placeholderImage} style={[styles.image, {width: size, height: size}]} />
+          <Spacer height={IMAGE_PADDING} />
+          <Text style={[styles.textTop, {width: size}]}>{text}</Text>
+          <Spacer height={TEXT_PADDING} />
         </>
-      )}
+      ) : null}
     </TouchableOpacity>
```

Comment:
can we branch earlier this logic, like return <ApprovedNFT>, return <BlockedNFT>, or just arrange better this logic?